.github/scripts/hermetic_library_generation.sh

```diff
@@ -73,31 +81,23 @@ git checkout "${current_branch}"
 # copy generation configuration from target branch to current branch.
 git show "${target_branch}":"${generation_config}" > "${baseline_generation_config}"
 
-generator_version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout -pl gapic-generator-java)
-echo "Local generator version: ${generator_version}"
-
-# install generator locally since we're using a SNAPSHOT version.
-mvn -V -B -ntp clean install -DskipTests
+# get .m2 folder so it's mapped into the docker container
+m2_folder=$(dirname "$(mvn help:evaluate -Dexpression=settings.localRepository -q -DforceStdout)")
 
-# build image locally since we want to include latest change.
-docker build \
-  -f .cloudbuild/library_generation/library_generation.Dockerfile \
-  -t gcr.io/cloud-devrel-public-resources/java-library-generation:"${image_tag}" \
-  .
 # run hermetic code generation docker image.
 docker run \
   --rm \
   -u "$(id -u):$(id -g)" \
   -v "$(pwd):${workspace_name}" \
-  -v "$HOME"/.m2:/home/.m2 \
-  -e GENERATOR_VERSION="${generator_version}" \
+  -v "${m2_folder}":/home/.m2 \
```

Comment:
Do we still need to volume m2 folder now that generator is baked in?

Reply:
I'll try to remove it.

Reply:
So .m2 is still needed?

Reply:
If I removed this line, the build failed, please see https://github.com/googleapis/sdk-platform-java/pull/3173#issuecomment-2341588555

Reply:
Looks like this is a special case of sdk-platform-java because it depends on snapshot jars

Reply:
From offline discussion, we are dependent on [mvn fmt:format](https://github.com/googleapis/sdk-platform-java/blob/7996aabf90d1b5a35ad1ff32afab3c522572d959/library_generation/owlbot/bin/entrypoint.sh#L83), hence a local .m2 volumn is needed. This will be baked into the docker image soon.